src/app/pages/quickMaps/pages/projection/projectionFoodSources/projectionFoodSources.component.ts

```diff
@@ -37,52 +93,294 @@ export class ProjectionFoodSourcesComponent implements AfterViewInit {
+                .filter((item) => item.year === currentYear)
+                .filter((item) => item.commodity === 'Other')
+                .map((item) => item.value)[0],
+            },
+          });
+
+          //  Populate data for Table
+          const foodTypes = [...new Set(data.map((item) => item.commodity))];
+
+          foodTypes.forEach((thing, index) => {
+            filteredTableData.push({
+              year: currentYear,
+              foodName: foodTypes[index],
+              value: filteredByScenario
+                .filter((item) => item.year === currentYear)
+                .filter((item) => item.commodity === foodTypes[index])
+                .map((item) => item.value)[0],
+            });
+          });
+          filteredTableDataArray.push(filteredTableData);
+        });
+
+        this.errorSrc.next(false);
+        this.chartData = null;
+        // force change detection to:
+        // remove chart before re-setting it to stop js error
+        this.cdr.detectChanges();
+        this.initialiseGraph(filteredData);
+
+        // show table and init paginator and sorter
+        this.initialiseTable(filteredTableDataArray, this.currentYearId);
+      })
+      .catch((err) => {
+        this.errorSrc.next(true);
+        console.error(err);
+      })
+      .finally(() => {
+        this.loadingSrc.next(false);
+        this.cdr.detectChanges();
+      });
+  }
+
+  private initialiseTable(data: Array<ProjectedFoodSourcesTable>, yearId: number): void {
+    // Data needs to be flattened frmo a nested array as mat-table can't acccess nested data
+    const flatten = (arr: Array<ProjectedFoodSourcesTable>) =>
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      arr.reduce(
+        (a: ProjectedFoodSourcesTable[], b: ProjectedFoodSourcesTable) => a.concat(Array.isArray(b) ? flatten(b) : b),
+        [],
       );
-    } else if (null != this.dialogData) {
-      // if displayed within a dialog use the data passed in
-      // this.init(Promise.resolve(this.dialogData.dataIn.data));
-      // this.tabGroup.selectedIndex = this.dialogData.dataIn.selectedTab;
-      // this.cdr.detectChanges();
-    }
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const flattenedData = flatten([data[yearId]]);
+    this.dataSource = new MatTableDataSource(flattenedData);
+    this.dataSource.sort = this.sort;
   }
 
-  private openDialog(): void {
-    void this.dialogService.openDialogForComponent<ProjectionFoodSourcesDialogData>(ProjectionFoodSourcesComponent, {
-      // data: this.data,
-      // selectedTab: this.tabGroup.selectedIndex,
-    });
+  private initialiseGraph(data: Array<ProjectedFoodSourcesPeriod>): void {
+    const quinquennialPeriod = [...new Set(data.map((item) => item.year))];
+
+    this.chartData = {
+      type: 'bar',
+      data: {
+        labels: quinquennialPeriod,
+        datasets: [
+          {
+            label: 'Cassava',
```

Comment:
Same comment as above about it being nice to loop through these without hard-coded strings.

Reply:
done